server.js

```diff
@@ -0,0 +1,309 @@
+              'elementSymbol',
+              'elementAtomicNumber',
+              'elementDescription'
+            ];
+            let hasRequiredFields = true;
+            let requestBody = '';
+            let parsedRequestBody = '';
+            let htmlFileText = '';
+
+            req.on('data', chunk => {
+              requestBody += chunk;
+            });
+
+            req.on('end', chunk => {
+              if (chunk) {
+                requestBody += chunk;
+              }
+
+              parsedRequestBody = querystring.parse(requestBody);
+
+              requiredFields.forEach(field => {
+                if (!Object.keys(parsedRequestBody).includes(field)) {
+                  hasRequiredFields = false;
+                }
+              });
+
+              if (!hasRequiredFields) {
+                res.writeHead(406);
+                res.end('request does not provide required fields');
+              } else {
+                htmlFileText = `<!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8" />
+            <title>The Elements - Helium</title>
+            <link rel="stylesheet" href="/css/styles.css" />
+          </head>
+          <body>
+            <h1>${parsedRequestBody.elementName}</h1>
+            <h2>${parsedRequestBody.elementSymbol}</h2>
+            <h3>Atomic number ${parsedRequestBody.elementAtomicNumber}</h3>
+            <p>
+              ${parsedRequestBody.elementDescription}
+            </p>
+            <p><a href="/">back</a></p>
+          </body>
+        </html>
+        `;
+
+                fs.writeFile(
+                  `./public${req.url}`,
+                  htmlFileText,
+                  (err, data) => {
+                    if (err) {
+                      res.writeHead(500, {
+                        'Content-Type': 'application/JSON'
+                      });
+                      res.end('{ "error": "could not write file" }');
+                    } else {
+                      res.writeHead(200, {
+                        'Content-Type': 'application/JSON'
+                      });
+                      res.end('{ "success": true }');
+                    }
+                  }
+                );
+              }
+            });
+          }
+        });
+      }
+
+      if (req.method === 'DELETE') {
+        fs.access('./public' + req.url, err => {
+          if (err) {
+            res.writeHead(500, { 'Content-Type': 'application/JSON' });
+            res.end(`{ "error": "resource ${req.url} does not exist" }`);
+          } else {
+            let indexHTMLText = '';
+
+            fs.unlink('./public' + req.url, err => {
+              if (err) {
+                res.writeHead(500, { 'Content-Type': 'application/JSON' });
+                res.end('{ "success": false, could not delete file }');
+              }
+            });
+
+            fs.readFile('./public/index.html', 'utf8', (err, data) => {
```

Comment:
This readFile should be in the unlink above or else you could be updating the index file and removing a file that actually still exists.